api/src/main/java/com/datastrato/gravitino/authorization/SecurableObjects.java

```diff
@@ -17,10 +20,12 @@ public class SecurableObjects {
   /**
    * Create the {@link SecurableObject} with the given names.
    *
+   * @param type The securable object type.
    * @param names The names of the securable object.
    * @return The created {@link SecurableObject}
    */
-  public static SecurableObject of(String... names) {
+  public static SecurableObject of(SecurableObject.Type type, String... names) {
```

Comment:
This names maybe errorprone, you require that names be ordered like catalog, schema, table, but in your code you cannot do such enforcement, so I would suggest you redefine your interface to avoid misuse.

Reply:
I removed the public.